src/ServiceControl/Persistence/PersistenceFactory.cs

```diff
@@ -49,5 +54,36 @@ public static AssemblyLoadContext DetermineLoadContext(Settings settings, string
                 ? new PluginAssemblyLoadContext(assemblyPath)
                 : loadContext;
         }
+
+        public static object LoadComponentPersistence(Settings settings, string componentPersistenceAssemblyPath, string componentPersistenceType)
+        {
+            if (!loadedComponentPersisters.Contains(componentPersistenceType))
+            {
+                AssemblyLoadContext.Default.Resolving += ResolvePersistenceAssemblyInDefaultContext;
+            }
+
+            Assembly ResolvePersistenceAssemblyInDefaultContext(AssemblyLoadContext context, AssemblyName name) =>
```

Comment:
Hmm I just realized looking at this that I actually have a bug here.  It actually should be something like

```suggestion
            static Assembly ResolvePersistenceAssemblyInDefaultContext(AssemblyLoadContext context, AssemblyName name)
            {
                var path = Path.Combine(<componentPersistenceLocation>, $"{name.Name}.dll");

                return File.Exists(path)
                ? context.LoadFromAssemblyPath(path)
                : null;
            }
```

Reply:
Because it is returning the component's persistence assembly regardless of which assembly it was actually trying to resolve (given by the AssemblyName parameter).  

This will fail if it is trying to locate a dependency of our persistence assembly, since we would actually return the persistence assembly itself.  Also while debugging I have seen instances where it is trying to resolve assemblies that are unrelated to our persistence seam, and in those cases we should be returning null, instead of loading our persistence assembly.